src/NLog/LogFactory.cs

```diff
@@ -142,25 +145,39 @@ internal LogFactory(ILoggingConfigurationLoader configLoader)
         /// </summary>
         public static IAppDomain CurrentAppDomain
         {
-            get
-            {
-                return currentAppDomain ??
-#if NETSTANDARD1_0
-                    (currentAppDomain = new FakeAppDomain());
-#else
-                    (currentAppDomain = new AppDomainWrapper(AppDomain.CurrentDomain));
-#endif
-            }
+            get => currentAppDomain ?? DefaultAppEnvironment.AppDomain;
```

Comment:
Idem, why not CurrentAppEnvironment?

Reply:
Because CurrentAppDomain is a static-member (Should be called DefaultAppDomain as it is global). And cannot reference a member-variable